### DISCUSSION TITLE
How to Access the Item States.

### DISCUSSION THREAD

**Discussion by LukasOchmann** [rpldy/react-uploady#136] [Q&A]
I currently need to Access the Item States, so that i can display an Icon per state.

My current solution is this custome hook: 
```ts

export default function useItemState(item: BatchItem): FILE_STATES {
  const [state, setState] = useState<FILE_STATES>(FILE_STATES.ADDED);
  const id = item.id;

  useItemStartListener(() => setState(FILE_STATES.UPLOADING), id);
  useItemFinishListener(() => setState(FILE_STATES.FINISHED), id);
  useItemErrorListener(() => setState(FILE_STATES.ERROR), id);
  useItemCancelListener(() => setState(FILE_STATES.CANCELLED), id);
  useItemAbortListener(() => setState(FILE_STATES.ABORTED), id);

  return state;
}
```
But if i abort one of the current uploads, the Listener get lost. And the States did not update.

I will try to create an sandbox.

**Comment by yoavniran** [comment]
Hey @LukasOchmann 
not sure what you mean by "get lost"

a sandbox will be very useful here. For example I want to understand how you get the item instance that you pass to the custom hook

**Reply by LukasOchmann** [reply]
Hey @yoavniran,
While uploading multiple files and just cancle one, It appears that the hook above wont update for the remaining files/items. In the devtools i see the uploads finishing successfully. 
But this only appears when the files are added in 2 (ore more) Batches.

I collect all items into an React.context using the useBatchAddListener-hook.

Here the Sandbox: 
https://codesandbox.io/s/pensive-fermi-2w5cq?fontsize=14&hidenavigation=1&theme=dark
But i could not reproduce the issue ...

**Reply by yoavniran** [reply]
thanks for the sandbox. It does seem to reproduce for me 
I will investigate and let you know

**Reply by yoavniran** [reply]
I identified the issue. Its related to the way batches are loaded into the queue when concurrent uploads are allowed.
I should have a fix in the next couple of days

**Reply by LukasOchmann** [reply]
Thank you for your time and help.

**Comment by yoavniran** [comment]
version 0.10.0 just released. 
Should fix the issue
